File: tests/fixtures/gold/typescript/public_surface.ts
export function add(a: number, b: number): number {
  return a + b;
}

export class Router {
  handle(): void {}
}

export interface Repo {
  save(): void;
}

export type Point = { x: number; y: number };

export enum Color {
  Red,
}

export const VERSION: string = "1.0";

File: tests/fixtures/gold/zig/public_surface.zig
const std = @import("std");

pub const VERSION = "0.1.0";

pub const Point = struct {
    x: f32,
    y: f32,

    pub fn distance(a: Point, b: Point) f32 {
        _ = a;
        _ = b;
        return 0;
    }

    fn reset(self: *Point) void {
        _ = self;
    }
};

pub const Color = enum {
    red,
    green,
};

pub fn greet(name: []const u8) void {
    _ = name;
}

pub var counter: u32 = 0;

fn helper() void {}
